### ISSUE TITLE
Compilation errors on Elixir 1.6.0

### ISSUE THREAD

**Issue by lleger** [appsignal/appsignal-elixir#296]
I upgraded my server to Elixir 1.6 today and started seeing errors with Appsignal:

```bash
$ iex -S mix
Erlang/OTP 20 [erts-9.2] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:10] [hipe] [kernel-poll:false]

01:05:50.490 [error] Failed to start AppSignal. Please run the diagnose task (https://docs.appsignal.com/elixir/command-line/diagnose.html) to debug your installation.
Interactive Elixir (1.6.0) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> 
```

Running the diagnose command, everything looked fine, but I did see this error:

```bash
$ mix appsignal.diagnose
[2018-01-18 19:23:10.337166Z] Error loading NIF (Is your operating system (x86_64-unknown-linux-gnu) supported? Please check http://docs.appsignal.com/support/operating-systems.html):
Failed to load NIF library: '/tmp/build_9888e7d4191f220c093a5f9b4eb0ceaf/_build/prod/lib/appsignal/priv/appsignal_extension.so: cannot open shared object file: No such file or directory'

[2018-01-19 01:00:06.088089Z] Error loading NIF (Is your operating system (x86_64-unknown-linux-gnu) supported? Please check http://docs.appsignal.com/support/operating-systems.html):
Failed to load NIF library: '/app/_build/prod/lib/appsignal/priv/appsignal_extension.so: cannot open shared object file: No such file or directory’
```

Out of curiosity, I tried to compile deps again:

```bash
$ mix deps.compile 
==> appsignal
could not compile dependency :appsignal, "mix compile" failed. You can recompile this dependency with "mix deps.compile appsignal", update it with "mix deps.update appsignal" or clean it with "mix deps.clean appsignal"
==> my_app
** (Mix) Could not run `make`. Please check if `make` and either `clang` or `gcc` are installed
```

This didn’t happen on Elixir 1.5.x. This doesn’t happen on my local machine (Mac OS X) but does happen on the server (Heroku).

**Comment by lleger** [comment]
Searched around some more, found #273 and can report this:

```bash
$ iex -S mix
iex(1)> :erlang.system_info(:system_architecture)
'x86_64-unknown-linux-gnu’
```

However, setting `APPSIGNAL_BUILD_FOR_MUSL=1` did not resolve this for me.

**Comment by tombruijn** [comment]
Hi @lleger ,

That's not good! Looks like the build tools are not available. It cannot find `make`. This happens on Heroku? Are you using a buildpack and if so, which one?
And what version of AppSignal are you running?

**Comment by lleger** [comment]
`mix.lock`:

```
"appsignal": {:hex, :appsignal, "1.4.5", "38309b8da51c728a282c1a8574deb2e7a9bb0a25c90624d85a93b4a98f136f7e", [], [{:decorator, "~> 1.2.2", [hex: :decorator, repo: "hexpm", optional: false]}, {:httpoison, "~> 0.11", [hex: :httpoison, repo: "hexpm", optional: false]}, {:phoenix, ">= 1.2.0", [hex: :phoenix, repo: "hexpm", optional: true]}, {:plug, ">= 1.1.0", [hex: :plug, repo: "hexpm", optional: true]}, {:poison, ">= 1.3.0", [hex: :poison, repo: "hexpm", optional: false]}], "hexpm"},
```

Buildpack: https://github.com/HashNuke/heroku-buildpack-elixir

Note that downgrading to 1.5.3 fixed the problem.

**Comment by jeffkreeftmeijer** [comment]
This was reported again via [support](https://app.intercom.io/a/apps/yzor8gyw/respond/inbox/540654/conversations/14163162539) (private Intercom link). This shouldn't be an issue caused by `make` missing, as downgrading to 1.5.3 fixed the problem for @lleger. We'll do some testing to see what's going wrong here, and we'll update this issue when we find more.

Thanks for letting us know about this, Logan!

**Comment by lleger** [comment]
This was fixed by #298.
